tests/Export/XmlExportTest.php

```diff
@@ -98,4 +100,33 @@ protected function getExport(): XmlExport
             $this->crossSellCategories
         );
     }
+
+    public function testProductPriceWithCurrency(): void
+    {
+        $currencyId = $this->createCurrency();
+        $this->salesChannelContext->getSalesChannel()->setCurrencyId($currencyId);
+        $this->getContainer()->set('fin_search.sales_channel_context', $this->salesChannelContext);
+        $testProduct = $this->createTestProduct([
+            'price' => [
+                ['currencyId' => Defaults::CURRENCY, 'gross' => 15, 'net' => 10, 'linked' => false],
+                ['currencyId' => $currencyId, 'gross' => 7.5, 'net' => 5, 'linked' => false]
+            ]
+        ]);
+
+        $customerGroupEntities = $this->getContainer()
+            ->get('customer_group.repository')
+            ->search(new Criteria(), $this->salesChannelContext->getContext())
+            ->getElements();
+
+        $items = $this->getExport()->buildItems(
+            [$testProduct],
+            self::VALID_SHOPKEY,
+            $customerGroupEntities
+        );
+
+        $item = $items[0];
+        $price = $item->getPrice();
+        $priceValues = $price->getValues();
+        $this->assertEquals(7.5, current($priceValues));
```

Comment:
How can this pass for versions below 6.4, if you don't set a currency id?

Reply:
Because we do not necessarily need the currency ID from the context, as you pointed out in the previous comment. In the code we take the currency ID from the sales channel, which is why the tests pass because we do not need the context currency ID. Have fixed that.